kernel/CMakeLists.txt

```diff
@@ -117,6 +117,35 @@ target_sources_ifdef(CONFIG_OBJ_CORE              kernel PRIVATE obj_core.c)
 
 if(${CONFIG_KERNEL_MEM_POOL})
   target_sources(kernel PRIVATE mempool.c)
+
+  # Import all custom HEAP_MEM_POOL minimum size requirements
+  import_kconfig(CONFIG_HEAP_MEM_POOL_MIN_SIZE_ ${DOTCONFIG} min_size_keys)
+
+  set(max_min_size 0)
+
+  foreach(min_size ${min_size_keys})
+    if(${${min_size}} GREATER ${max_min_size})
+      set(max_min_size ${${min_size}})
+    endif()
+  endforeach()
+
+  if(CONFIG_HEAP_MEM_POOL_SIZE LESS max_min_size)
```

Comment:
when testing, then it is allowed to use the variable name directly like this, but in the lines above you where de-referencing, like `GREATER ${max_min_size}`.

So for consistency, maybe dereference `max_min_size` here also.

Reply:
Done